File: Ejercicios Java/Diccionario.java
import java.util.HashMap;

public class Diccionario {
    public static void main(String[] args) {
        // Definimos un HashMap, que es una colección de pares clave:valor
        HashMap<String, String> coche = new HashMap<>();

        // Agregamos atributos al HashMap
        coche.put("marca", "Ford");
        coche.put("color", "rojo");
        coche.put("modelo", "sedan");
        coche.put("placa", "ROS227");

        // Imprime todos los atributos contenidos en "coche"
        System.out.println(coche);

        // Imprime el atributo "color"
        System.out.println(coche.get("color"));

        // Cambia el valor dentro del atributo "color"
        coche.put("color", "verde");

        // Imprime el nuevo valor
        System.out.println(coche.get("color"));

        // Cambia el valor dentro del atributo "marca"
        coche.put("marca", "Renault");

        // Imprime el nuevo valor
        System.out.println(coche.get("marca"));

        // Imprime todos los atributos contenidos en "coche" luego de cambios
        System.out.println(coche);
    }
}

File: Ejercicios_Rubi/Diccionario.rb
# Diccionario en Ruby con formato literal
auto = {'marca' => 'Ford', 'color' => 'rojo', 'modelo' => 'sedan', 'placa' => 'ROS227'}

# Diccionario en Ruby con símbolos como claves
coche = {:marca => 'Ford', :color => 'rojo', :modelo => 'sedan', :placa => 'ROS227'}

# Imprimimos el elemento de la clave :color en el hash coche
puts coche[:color]  # 'rojo'

# Modificamos el elemento de la clave :color en el hash coche
coche[:color] = "Verde"
# Imprimimos el nuevo valor de la clave :color
puts coche[:color]  # 'Verde'

# Ahora con la marca
puts coche[:marca]  # 'Ford'
coche[:marca] = "Renault"
puts coche[:marca]  # 'Renault'

# Recorrer el diccionario y traer todos los elementos con .each
coche.each do |clave, valor|
  puts "#{clave}: #{valor}"  # Esto imprimirá todos los pares clave-valor del diccionario
end
